src/SignalR/server/Core/src/HubConnectionContext.cs

```diff
@@ -355,15 +381,15 @@ private async Task WriteHandshakeResponseAsync(HandshakeResponseMessage message)
         /// </summary>
         public virtual void Abort()
         {
+            _connectionAborted = true;
```

Comment:
Why was this moved? If _connectionAbortedTokenSource.IsCancellationRequested is true, doesn't that mean that Abort() must have been called?

Also, since _connectionAborted is set outside a lock and PipeReader.CancelPendingRead doesn't necessarily schedule the reader because the reader can observer cancellation synchronously, _connectionAborted should be `volatile`.

Reply:
Forget about the `volatile`. My brain is still a little poisoned back from when Pipes/SocketInput locked less. [This lock](https://github.com/dotnet/corefx/blob/87b2e237fafbdfb01f03f16558f3fa17f8288f91/src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs#L600) should mean the volatile is unnecessary.